botocore/parsers.py

```diff
@@ -1020,19 +1035,29 @@ def _inject_error_code(self, error, response):
         # The "Code" value can come from either a response
         # header or a value in the JSON body.
         body = self._initial_body_parse(response['body'])
+        code = None
         if 'x-amzn-errortype' in response['headers']:
             code = response['headers']['x-amzn-errortype']
-            # Could be:
-            # x-amzn-errortype: ValidationException:
-            code = code.split(':')[0]
-            error['Error']['Code'] = code
         elif 'code' in body or 'Code' in body:
-            error['Error']['Code'] = body.get('code', body.get('Code', ''))
+            code = body.get('code', body.get('Code', ''))
+        if code is not None:
+            if ':' in code:
+                code = code.split(':', 1)[0]
+            if '#' in code:
+                code = code.split('#', 1)[1]
+            error['Error']['Code'] = code
```

Comment:
It looks like we're changing behavior again here. Is there additional scope, or why are we post-processing the body for the `code` case now?